contracts/src/v0.8/automation/dev/v2_3/AutomationRegistryLogicB2_3.sol

```diff
@@ -130,9 +130,10 @@ contract AutomationRegistryLogicB2_3 is AutomationRegistryBase2_3 {
     if (s_upkeepAdmin[id] != msg.sender) revert OnlyCallableByAdmin();
     if (upkeep.maxValidBlocknumber > s_hotVars.chainModule.blockNumber()) revert UpkeepNotCanceled();
     uint96 amountToWithdraw = s_upkeep[id].balance;
-    s_reserveAmounts[address(i_link)] = s_reserveAmounts[address(i_link)] - amountToWithdraw;
+    s_reserveAmounts[address(upkeep.billingToken)] = s_reserveAmounts[address(upkeep.billingToken)] - amountToWithdraw;
     s_upkeep[id].balance = 0;
-    i_link.transfer(to, amountToWithdraw);
+    bool success = upkeep.billingToken.transfer(to, amountToWithdraw);
```

Comment:
i think `transfer` may return false or just revert depending on if the token's contract complies with ERC20 specs.
i think billing token belongs to a small subset of allowed tokens?

Reply:
tbh I don't know of any situations where `transfer()` would return false but not revert. But it felt lazy to just ignore the result so I included the require here. And yes, billing token is a small permissioned list of allowed tokens.